### DISCUSSION TITLE
Self-Hosted composer registry on GitLab gives wrong package url

### DISCUSSION THREAD

**Discussion by mrgesco** [renovatebot/renovate#10646] [Request Help] [answered]
Hi,

I'm using the self-hosted version of renovate on my GitLab instance.
I have created a Composer registry on my GitLab and trying to use two composer repositories on a project.

When renovate parses the composer file and tries to check the version, it gives a 404 error while trying to check on packagist instead of the right repository url.

Have you any idea how to set the correct one in renovate?

composer.json
```
...
"repositories": [
    {
        "type": "composer",
        "url": "https://wpackagist.org",
        "only": ["wpackagist-plugin/*", "wpackagist-theme/*"]
    },
    {
        "type": "composer",
        "url": "https://***/api/v4/group/**/-/packages/composer/packages.json"
    }
],
"require": {
    "***REDACTED***" : "5.9.6"
}
```

Renovate Debug
```
DEBUG: Datasource 404 (repository=***REDACTED***)
       "datasource": "packagist",
       "lookupName": "***REDACTED***",
       "url": "https://packagist.org/p/***REDACTED***.json"
DEBUG: Failed to look up dependency ***REDACTED*** (repository=***REDACTED***, packageFile=composer.json, dependency=***REDACTED***)
....
{
    "depType": "require",
    "depName": "***REDACTED***",
    "currentValue": "^5.9",
    "datasource": "packagist",
    "lockedVersion": "5.9.6",
    "depIndex": 18,
    "updates": [],
    "warnings": [
    {
        "topic": "***REDACTED***",
        "message": "Failed to look up dependency ***REDACTED***"
    }
    ]
}
```

**Comment by rarkins** [comment, marked as answer]
Please file it as a bug issue with a minimal reproduction repo. It looks like the composer.json may not be parsed fully. URLs do not need to be *exact* what you have but should be equivalent.

**Reply by mrgesco** [reply]
Unfortunately my GitLab repository is private and I can't therefore create a reproduction repo...

**Reply by rarkins** [reply]
Please re-read my comment above about the URLs not needing to be exact. We just need it to fail the same way.

**Reply by mrgesco** [reply]
Oh ok sorry I'm going to create this.

Thank you.
